src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -120,6 +190,30 @@ public static Complex Divide(Complex dividend, Complex divisor)
             }
         }
 
+        public static Complex operator /(Complex left, double right)
+        {
+            return new Complex(left.m_real / right, left.m_imaginary / right);
+        }
+
+        public static Complex operator /(double left, Complex right)
+        {
+            // Division : Smith's formula.
+            double a = left;
+            double c = right.m_real;
+            double d = right.m_imaginary;
+
+            if (Math.Abs(d) < Math.Abs(c))
+            {
+                double doc = d / c;
+                return new Complex(a / (c + d * doc), (-a * doc) / (c + d * doc));
```

Comment:
I think this algorithm can be simplified (and made a little more performant).

```C#
            //  a        a     c-di    ac      -ad
            // ----   = ---- X ---- = ----- + ---- i
            // c+di     c+di   c-di   cc+dd   cc+dd
            double a = left;
            double c = right.Real;
            double d = right.Imaginary;

            double divisor = (c * c) + (d * d);
            return new Complex((a * c) / divisor, -(a * d) / divisor);
```

Check out this gist:  https://gist.github.com/eerhardt/91698ac44abdb298ce6c693bb27299cd

The Benchmark numbers I'm getting on my machine are:

```
BenchmarkDotNet=v0.10.14, OS=Windows 10.0.17134
Intel Core i7-6700 CPU 3.40GHz (Skylake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=2.1.400-preview-008942
  [Host]     : .NET Core 2.1.0-rtm-26515-03 (CoreCLR 4.6.26515.03, CoreFX 4.6.26515.03), 64bit RyuJIT
  DefaultJob : .NET Core 2.1.0-rtm-26515-03 (CoreCLR 4.6.26515.03, CoreFX 4.6.26515.03), 64bit RyuJIT
```

|     Method |     Mean |     Error |    StdDev |
|----------- |---------:|----------:|----------:|
|   Original | 81.41 ns | 1.1368 ns | 1.0634 ns |
| Simplified | 64.99 ns | 0.7056 ns | 0.6600 ns |

Reply:
I haven't reviewed carefully, but I think this proposed change is dangerous in precisely the way Smith's algorithm is coded to avoid. Computing c*c + d*d will overflow even in cases where the actual result of the division does not overflow.

Reply:
Good point on the overflow. For sure adding a code comment here describing why the code is written this way would be helpful.